google-cloud-bigquerystorage/src/main/java/com/google/cloud/bigquery/storage/v1alpha2/StreamWriter.java

```diff
@@ -265,6 +273,28 @@ public Boolean expired() {
     return outstandingAppend.appendResult;
   }
 
+  /**
+   * This is the general flush method for asynchronise append operation. When you have outstanding
+   * append requests, calling flush will make sure all outstanding append requests completed and
+   * successful. Otherwise there will be an exception thrown.
+   *
+   * @throws Exception
+   */
+  public void flushAll() throws Exception {
+    writeAllOutstanding();
+    synchronized (messagesWaiter) {
```

Comment:
What happens if user still keep appending when we call flushAll? Should we also wait for the following append requests?

Reply:
It is on them...now we have timeout, hopefully it will timeout.

Reply:
Good point. I added a lock around these operations so no append can happen in between and cause it to hault.

Reply:
I had this discussion with Bigang previously and we decided to keep allow user to append instead of block. Just wait for all pending stream before the time user call FlushAll() but keep accepting new requests. (Flush all will not wait for new requests).

Do you think this make sense to this case as well?